### PR TITLE
NR-827 Convert scripts to TypeScript

### PR DESCRIPTION
**Pull request by mickeyryan42**
# Summary

Converting scripts to typescript
https://issues.newrelic.com/browse/NR-827

### CODE REVIEW

File: utils/check_quickstart_uniqueness.ts [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
-import type { FilePathAndContents } from './helpers';
-
-import {
+'use strict';
+const {
   readQuickstartFile,
   removeRepoPathPrefix,
   findMainQuickstartConfigFiles,
-} from './helpers';
+} = require('./helpers');
```
Comment (josephgregoryii) [inline utils/check_quickstart_uniqueness.ts:6]: I think it might be more consistent if we use imports here instead of require.

Comment (josephgregoryii) [inline utils/check_quickstart_uniqueness.ts:6]: I think it might be more consistent if we use imports here instead of require.

**Review by josephgregoryii — CHANGES_REQUESTED** [review]
Looks great! just a few comments/questions

File: utils/check_quickstart_uniqueness.ts (hunk 1 above)
Comment (mickeyryan42) [reply utils/check_quickstart_uniqueness.ts:6]: Strange i dont remember doing this, and not sure why i did.  Let me go switch it back

File: utils/check_quickstart_uniqueness.ts [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
+export type IdsAndPaths = {
+  id: string;
+  path: string;
+};
 /**
  * Returns any quickstarts with matching ids
- * @param idsAndPaths - an array of objects containing the path and id of a quickstart
- * @returns - an array of matching values
+ * @param {Object[]} idsAndPaths an array of objects containing the path and id of a quickstart
+ * @returns {Object[]} an array of matching values
```
[the remark this answers is not in the document]
Comment (mickeyryan42) [reply utils/check_quickstart_uniqueness.ts:18]: I meant to ask whether we wanted to remove it. Will do

File: utils/check_quickstart_uniqueness.ts [hunk 3]
```diff
@@ -1,26 +1,24 @@
-import type { QuickstartConfig } from './types/QuickstartConfig';
-import type { FilePathAndContents } from './helpers';
-
+'use strict';
```
Comment (aswanson-nr) [inline utils/check_quickstart_uniqueness.ts:1]: We can remove this now that it's a typescript file, the compilation step handles setting strict mode for us!

File: utils/generate-uuids.ts [hunk 4]
```diff
@@ -2,20 +2,23 @@
...
 const fs = require('fs');
 const { v4: uuidv4 } = require('uuid');
-const { findMainQuickstartConfigFiles, readYamlFile } = require('./helpers');
+import { findMainQuickstartConfigFiles, readYamlFile } from "./helpers";
+
+import { QuickstartConfig } from './types/QuickstartConfig';
 
 /**
  * Checks for and generates UUIDs for quickstarts
  * @param {String[]} paths an array of quickstarts config paths
```
Comment (josephgregoryii) [inline utils/generate-uuids.ts]: Lets go ahead and remove the types in the JSDocs in this PR 👍🏻

File: utils/check_quickstart_uniqueness.ts [hunk 5]
```diff
@@ -67,4 +61,4 @@ if (require.main === module) {
   main();
 }
 
-module.exports = main;
+module.exports = main;
```
Comment (aswanson-nr) [inline utils/check_quickstart_uniqueness.ts:67]: Let's update this export now that we have typescript!
We can totally remove this line
```suggestion
```
And update the `const main = () =>` function to be `export default const main = () =>`

**Review by aswanson-nr — COMMENTED** [review]
Just a few small cleanup comments around exporting!

**Review by josephgregoryii — CHANGES_REQUESTED** [review]
Just one last think and we should be good to go :)

File: utils/check-missing-logos.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
+const {
+  readQuickstartFile,
+  findMainQuickstartConfigFiles,
+} = require('./helpers');
```
Comment (josephgregoryii) [inline utils/check-missing-logos.ts:4]: Looks like we just have one more import that needs to be updated 😄

Comment (mickeyryan42) [reply utils/check-missing-logos.ts]: Lol thanks! They were really flying under my radar. I got it removed

**Review by josephgregoryii — APPROVED** [review]

**Comment by josephgregoryii** [comment]
Great work on this 😄

**Review by aswanson-nr — APPROVED** [review]

**Pull request merged**